### PR TITLE
Draw Rectangle Tool

### LINKED ISSUE #146: Draw rectangle
Hi, I saw there is already Circle support. Could this be also implemented for Rectangle? Thanks!

### PR DESCRIPTION
**Pull request by Mesonyx**
Adds a tool for drawing a Rectangle polygon.

Closes #146 

(Lots of logic/ideas are shamelessly "inspired" by [Leaflet Draw](https://github.com/Leaflet/Leaflet.draw))

A few notes:

- It enforces itself as a "true" rectangle, so the capability to add additional sides (as with a dragging a Polygon's vertex) has been removed. If you wish to have an editable Rectangle, just use the Polygon tool.

- I organized a few methods in a way I felt made sense, and was internally consistent with the rest of the program. If this becomes an issue, please feel free to request that I put `L.PM` back the way it was 😉 

**ALL KNOWN BUGS HAVE BEEN ADDRESSED!**

~~**Known Bugs** ~~

- ~~Occasionally, southwest marker will fail to place correctly when adjusting size while snapped to another Polygon~~

- ~~When a Rectangle "crosses over itself" (i.e. you drag the starting marker past its opposite corner's latitude or longitude) during resizing in Edit Mode, it fails to snap to other Polygons~~

### CODE REVIEW

**Comment by adamtLICOR** [comment]
I should hopefully have some more time to commit to this next week.

I think I have a fix for the vanishing marker, among other things.

**Comment by Mesonyx** [comment]
Looks like my fix for the snapping bug also addressed the disappearing marker bug.

That takes care of all the known bugs I'm aware of!

**Comment by Mesonyx** [comment]
@codeofsumit -- I've fixed the known bugs and merge conflicts!

Let me know if this PR requires anything else (and/or simply isn't up to standards for whatever reason :D )

**Comment by Mesonyx** [comment]
Alright, it's been adjusted to work with the new cutting tool! 😸 

screen shot 2017-09-01 at 4 05 29 pm

screen shot 2017-09-01 at 4 05 33 pm

File: src/js/Mixins/Snapping.js [hunk 1]
```diff
@@ -100,7 +100,7 @@ const SnapMixin = {
             const a = this._snapLatLng || {};
             const b = snapLatLng || {};
 
-            if(a.lat !== b.lat || b.lng !== b.lng) {
+            if(a.lat !== b.lat || a.lng !== b.lng) {
```
Comment (codeofsumit) [inline src/js/Mixins/Snapping.js]: haha, damn!

**Comment by codeofsumit** [comment]
@Mesonyx I really appreciate the work you've put into this. Thank you! It works like a charm.

**Pull request merged**